src/v/cloud_storage/tx_range_manifest.cc

```diff
@@ -12,20 +12,239 @@
 #include "bytes/iobuf.h"
 #include "bytes/iostream.h"
 #include "bytes/streambuf.h"
-#include "cloud_storage/partition_manifest.h"
 #include "cloud_storage/types.h"
 #include "container/fragmented_vector.h"
-#include "json/istreamwrapper.h"
 #include "model/record.h"
 
 #include <rapidjson/document.h>
+#include <rapidjson/error/en.h>
 #include <rapidjson/istreamwrapper.h>
 #include <rapidjson/ostreamwrapper.h>
 #include <rapidjson/rapidjson.h>
 #include <rapidjson/writer.h>
 
+#include <stdexcept>
+
 namespace cloud_storage {
 
+struct tx_range_manifest_json_handler {
+    using Ch = rapidjson::UTF8<>::Ch;
+
+    // rapidjson's Handler concept.
+    bool Null() { return false; }
+    bool Bool(bool) { return false; }
+    bool Int(int i) { return Int64(i); }
+    bool Uint(unsigned u) { return Int64(u); }
+    bool Int64(int64_t i64) {
+        switch (_state) {
+        case state::expect_version_value:
+            version = static_cast<int>(i64);
```

Comment:
Having the name of parameter resemble the name of a type makes this a bit hard for my eyes to parse. Maybe rename `i64` to something like `val`?